### PR TITLE
Wireshark

### PR DESCRIPTION
**Pull request by baentsch**
Building on https://github.com/open-quantum-safe/oqs-demos/pull/104 (thanks to @anhu) this is an OQS-enabled wireshark build-and-dockerize command set. This will remain in draft status until documentation has been refined and final contents are agreed.

For a quick test/avoiding the need to build it locally, the image is available at docker hub under the name "openquantumsafe/wireshark", i.e., it can be executed (on X11-based UNIX hosts) via 

```
docker run --net=host --privileged --env="DISPLAY" --volume="$HOME/.Xauthority:/root/.Xauthority:rw" openquantumsafe/wireshark
```

All feedback welcome (tagging explicity @anhu @dstebila ) and suggesting to continue the discussion from #104 below.

### CODE REVIEW

File: wireshark/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+     docker run -it openquantumsafe/curl sh -c "curl -k https://test.openquantumsafe.org:6120 --curves p521_frodo1344aes"
+
+For more details regarding the client-side options, we recommend reviewing https://hub.docker.com/repository/docker/openquantumsafe/curl
+
+### Focus on quantum-safe data traffic 
+
+In order to focus `wireshark` on the quantum safe TLS traffic generated by the above `curl` command, we recommend setting a wireshark display filter as follows
+
+     ip.addr == 149.81.106.123
```
Comment (dstebila) [inline wireshark/USAGE.md:31]: I suggest filtering by TLS to make it easier to see that traffic:

```suggestion
     ip.addr == 149.81.106.123 && tls
```

File: wireshark/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+The image is based on Ubuntu and requires the host to run the Unix X-Window system.
+
+## Quick start
+
+Execute this command to open the wireshark window on your host:
+
+    docker run --net=host --privileged --env="DISPLAY" --volume="$HOME/.Xauthority:/root/.Xauthority:rw" openquantumsafe/wireshark
+
+*Note:* Select a network interface to monitor.
+
```
Comment (dstebila) [inline wireshark/USAGE.md:16]: I had to do the following on Ubuntu to make it run:
```suggestion

You may need to grant permissions for Docker to access the X display:

    xhost +si:localuser:$USER

```

File: wireshark/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+## Quick start
+
+1) Be sure to have [docker installed](https://docs.docker.com/install).
+2) Run `docker build -t openquantumsafe/wireshark .` to create an QSC-enabled (codepoint and OID aware) wireshark docker image.
+
+## Usage
+
+Information how to use the image is [available in the separate file USAGE.md](USAGE.md).
+
```
Comment (dstebila) [inline wireshark/README.md:11]:
```suggestion

## Contributors

- Michael Baentsch
- Anthony Hu (wolfSSL)
```

**Review by dstebila — APPROVED** [review]
Also suggest adding Anthony Wu (wolfSSL) to /README.md.

**Comment by dstebila** [comment]
I was able to run it successfully on an Ubuntu VM. I couldn't get the settings right on my Mac, but it might be more complicated to get a GUI from Docker showing up on macOS' X host, so don't worry about it.

**Comment by anhu** [comment]
> Also suggest adding Anthony Wu (wolfSSL) to /README.md.

:smiley: Anthony Hu (wolfSSL)

(As there is no Anthony Wu at wolfSSL)

**Comment by dstebila** [comment]
> > Also suggest adding Anthony Wu (wolfSSL) to /README.md.
> 
> 😃 Anthony Hu (wolfSSL)
> 
> (As there is no Anthony Wu at wolfSSL)

I'm so sorry Anthony.

**Comment by anhu** [comment]
> I'm so sorry Anthony.
@dstebila , Please, don't worry about it.

File: wireshark/USAGE.md (hunk 1 above)
Comment (baentsch) [reply wireshark/USAGE.md:31]: Good improvement. Thx.

File: wireshark/USAGE.md (hunk 2 above)
Comment (baentsch) [reply wireshark/USAGE.md:16]: That should not be necessary: The .Xauthority volume mount should do the same trick. But then again, different distributions may do things differently... So added. Thx.

File: wireshark/README.md (hunk 3 above)
Comment (baentsch) [reply wireshark/README.md:11]: We so far only have a top-level contribution list (to which I added Anthony now) and think we should leave it at that in the interest of short README's if that's OK for you.

**Comment by baentsch** [comment]
> I was able to run it successfully on an Ubuntu VM. I couldn't get the settings right on my Mac, but it might be more complicated to get a GUI from Docker showing up on macOS' X host, so don't worry about it.

Thanks for the confirmation that it also works for you. If you should manage to find the time to get it to work, please add a PR with documentation how to do it for OSX.

**Comment by baentsch** [comment]
I'm puzzled: Everythings builds and runs OK when doing it on my local Ubuntu machine (or on an AWS VM), but fails in CCI:

```
AutoMoc: /opt/wireshark-3.4.9/ui/qt/capture_options_dialog.h:0: Note: No relevant classes found. No output generated.

AutoMoc subprocess error
------------------------
The moc process failed to compile
  "SRC:/ui/qt/utils/field_information.h"
into
  "SRC:/build/ui/qt/qtui_autogen/7HEHEGDA3T/moc_field_information.cpp"

Command
-------
/usr/lib/qt5/bin/moc -DG_DISABLE_DEPRECATED -DG_DISABLE_SINGLE_INCLUDES -DQT_CORE_LIB -DQT_GUI_LIB -DQT_MULTIMEDIA_LIB -DQT_NETWORK_LIB -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -I/opt/wireshark-3.4.9/build -I/opt/wireshark-3.4.9 -I/opt/wireshark-3.4.9/build/ui/qt -I/opt/wireshark-3.4.9/ui/qt -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/x86_64-linux-gnu/qt5 -I/usr/include/x86_64-linux-gnu/qt5/QtCore -I/usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -I/usr/include/x86_64-linux-gnu/qt5/QtMultimedia -I/usr/include/x86_64-linux-gnu/qt5/QtNetwork -I/usr/include/x86_64-linux-gnu/qt5/QtGui -I/usr/include/x86_64-linux-gnu/qt5/QtPrintSupport -I/usr/include/x86_64-linux-gnu/qt5/QtWidgets -I/usr/include -I/usr/include/c++/9 -I/usr/include/x86_64-linux-gnu/c++/9 -I/usr/include/c++/9/backward -I/usr/lib/gcc/x86_64-linux-gnu/9/include -I/usr/local/include -I/usr/include/x86_64-linux-gnu --include /opt/wireshark-3.4.9/build/ui/qt/qtui_autogen/moc_predefs.h -p ui/qt/utils -o /opt/wireshark-3.4.9/build/ui/qt/qtui_autogen/7HEHEGDA3T/moc_field_information.cpp /opt/wireshark-3.4.9/ui/qt/utils/field_information.h

Output
------
/opt/wireshark-3.4.9/ui/qt/utils/field_information.h:24: Error: Undefined interface
```

@anhu: Do you have an idea what's going on here? I'm tempted to drop this from CCI and move to Review as otherwise all is fine.

**Comment by anhu** [comment]
> @anhu: Do you have an idea what's going on here? I'm tempted to drop this from CCI and move to Review as otherwise all is fine.

I have to admit to ignorance as to what is going on here.  Sorry. 

BTW, Thank you very much for adding back the wolfssl naming convention via flavour!!  I'm very happy with that!

**Pull request merged**